packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -135,6 +135,49 @@ function initialize(obj, properties) {
 }
 
 /**
+  `CoreObject` is the base class for all Ember constructs. It establishes a
+  class system based on Ember's Mixin system, and provides the basis for the
+  Ember Object Model. `CoreObject` should generally not be used directly,
+  instead you should use `EmberObject`.
+
+  ## Usage
+
+  You can define a class by extending from `CoreObject` using the `extend`
+  method:
+
+  ```js
+  const Person = CoreObject.extend({
+    name: 'Tomster',
+  });
+  ```
+
+  For detailed usage, see the [Object Model](https://guides.emberjs.com/release/object-model/)
+  section of the guides.
+
+  ## Usage with Native Classes
+
+  Native Javascript `class` syntax can be used to extend from any `CoreObject`
+  based class:
+
+  ```js
+  class Person extends CoreObject {
+    init() {
+      super.init(...arguments);
+      this.name = 'Tomster';
+    }
+  }
+  ```
+
+  Some notes about `class` usage:
+
+  * Instead of using `this._super()`, you should use standard `super` syntax in
+    native classes. See the [MDN docs on classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Super_class_calls_with_super)
+    for more details.
+  * Avoid using the `constructor` in favor of `init`. Services are not available
+    during the `constructor` phase, and the nuances of the two can be confusing.
+  * Zebra-striping (using native classes, and switching back to the old Ember
```

Comment:
While I know we have used this term a bit, I don't think its very widely understood outside of our immediate circles. I'd prefer to remove the "zebra striping" phrasing and just use the parenthetical directly...